src/main/java/Lemon.java

```diff
@@ -0,0 +1,156 @@
+
+    enum Commands {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        EVENT,
+        DEADLINE,
+        DELETE
+    }
+    
+    public static void main(String[] args) {
+        Lemon lemon = new Lemon();
+        Scanner scan = new Scanner(System.in);
+        boolean run = true;
+
+        /** Program starts here **/
+        System.out.print(lemon.logoMsg);
+
+        while (run) {
+            System.out.println(lemon.barMsg);
+
+            String input = scan.next().toUpperCase();
+            System.out.println(lemon.barMsg);
+            try {
+                Commands command = Commands.valueOf(input);
+                switch (command) {
+                case BYE:
+                    run = false;
+                    break;
+                case LIST:
+                    if (lemon.numTasks == 0) System.out.println(lemon.emptyListMsg);
+                    else {
+                        System.out.println(lemon.listMsg);
+                        lemon.printList();
+                    }
+                    break;
+                case MARK: {
+                    int next = scan.nextInt();
+                    if (next > lemon.numTasks || next <= 0) {
+                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
+                    }
+
+                    System.out.println(lemon.markMsg);
+                    lemon.tasks.get(next - 1).markDone();
+
+                    System.out.println("   " + lemon.tasks.get(next - 1).toString());
+                    break;
+                }
+                case UNMARK: {
+                    int next = scan.nextInt();
+                    if (next > lemon.numTasks || next <= 0) {
+                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
+                    }
+
+                    System.out.println(lemon.unmarkMsg);
+                    lemon.tasks.get(next - 1).unmarkDone();
+
+                    System.out.println("   " + lemon.tasks.get(next - 1).toString());
+                    break;
+                }
+                case TODO: {
+                    String next = scan.nextLine();
+
+                    lemon.addNewTask(new Todo(next));
+                    break;
+                }
+                case DEADLINE: {
+                    String[] next = scan.nextLine().split("/by ");
+
+                    lemon.addNewTask(new Deadline(next[0], next[1]));
+                    break;
+                }
+                case EVENT: {
+                    String[] next = scan.nextLine().split("/from ");
+
+                    lemon.addNewTask(new Event(next[0], next[1]));
+                }
+                case DELETE: {
+                    int next = scan.nextInt();
+                    if (next > lemon.numTasks || next <= 0) {
+                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
+                    }
```

Comment:
This logic could be refactored out into a class that would parse the inputs. Clear what each switch-case does so good job!